<!DOCTYPE html>
<html lang="en"><head>
<script src="presentation_files/libs/clipboard/clipboard.min.js"></script>
<script src="presentation_files/libs/quarto-html/tabby.min.js"></script>
<script src="presentation_files/libs/quarto-html/popper.min.js"></script>
<script src="presentation_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="presentation_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="presentation_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="presentation_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="presentation_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.3.361">

  <meta name="author" content="Kelly Sovacool">
  <title>Improving machine learning models for microbiome analysis</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="presentation_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="presentation_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
  </style>
  <link rel="stylesheet" href="presentation_files/libs/revealjs/dist/theme/quarto.css">
  <link href="presentation_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="presentation_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="presentation_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="presentation_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
  
  <script src="presentation_files/libs/kePrint-0.0.1/kePrint.js"></script>
  <link href="presentation_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Improving machine learning models for microbiome analysis</h1>
  <p class="subtitle">and democratizing data science along the way</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Kelly Sovacool 
</div>
</div>
</div>

  <p class="date">Jun 22, 2023</p>
</section>
<section id="what-is-a-microbiome" class="slide level2">
<h2>What is a microbiome?</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/microbiome-definition.png" height="500"></p>
</figure>
</div>
<div class="footer">
<p>Berg <em>et al.</em> (2020). <a href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-020-00875-0">Microbiome</a></p>
</div>
</section>
<section id="the-human-gut-microbiome-changes-during-diseases" class="slide level2">
<h2>The human gut microbiome changes during diseases</h2>
<ul>
<li>Irritable bowel diseases</li>
<li>Colorectal cancer</li>
<li><em>Clostridioides difficile</em> infection</li>
</ul>
<div class="fragment">
<!-- -->
<h5 id="open-problems-and-opportunities">Open problems and opportunities</h5>
<ul>
<li>Understand the mechanistic role of the gut microbiome in gut diseases.</li>
<li>Identify biomarkers to improve diagnostic and prognostic tools.</li>
</ul>
</div>
<aside class="notes">
<ul>
<li>Changes in the taxonomic composition and metabolic activity of human microbiomes have been observed in several diseases including colorectal cancer (CRC) and Clostridioides difficile infection (CDI).</li>
<li><strong>CRC</strong> - enrichment for bacteria that produce enterotoxins in CRC gut microbiomes.</li>
<li><strong>CDI</strong> - treatment with antibiotics alters the taxonomic composition of the gut microbiome, allowing <em>C. diff</em> to colonize the gut.</li>
</ul>
<p>Possible causative role of the microbiome in disease etiology. mechanisms are not entirely understood. not always clear whether the disease causes the microbiome to change, the change in the microbiome causes the disease, or some combination.</p>
<ul>
<li><p>CDI: antibiotics disrupts the community and allows cdiff to colonize, but what causes some people to develop severe outcomes vs clear the infection easily?</p>
<ul>
<li>Bile acids metabolized by the gut bacteria can inhibit C. difficile growth and affect toxin production (4, 10, 11). Bacteria in the gut also can compete more directly with C. difficile through antibiotic production or nutrient consumption</li>
</ul></li>
<li><p>CRC: some CRC patients have strains of <em>E. coli</em> that produce enterotoxins, but not all.</p>
<ul>
<li>FIT cartridge sensitivity (true pos rate): 70% (early stage) to 80% (late stage)</li>
<li>colonoscopy sensitivity: 78% to 93%</li>
<li>detecting CRC early improves prognosis, but colonoscopies are invasive.</li>
</ul></li>
<li><p>Baxter et al: “The microbiota-based random forest model detected 91.7 % of cancers and 45.5 % of adenomas while FIT alone detected 75.0 % and 15.7 %, respectively. Of the colonic lesions missed by FIT, the model detected 70.0 % of cancers and 37.7 % of adenomas. We confirmed known associations of Porphyromonas assaccharolytica, Peptostreptococcus stomatis, Parvimonas micra, and Fusobacterium nucleatum with CRC. Yet, we found that the loss of potentially beneficial organisms, such as members of the Lachnospiraceae, was more predictive for identifying patients with adenomas when used in combination with FIT.”</p></li>
<li><p>However, these changes are not universal across individuals or datasets.</p></li>
<li><p>Mouse experiments indicate a possible causative role of the microbiome.</p>
<ul>
<li>fecal matter transplant from susceptible to germ-free mice</li>
<li>predict colonization following Cdiff challenge</li>
</ul></li>
<li><p>Open problems, and opportunities:</p>
<ul>
<li>Understand the mechanistic role of the gut microbiome in gut diseases.</li>
<li>Identify biomarkers to improve diagnostic and prognostic tools.</li>
</ul></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="the-human-gut-microbiome-changes-during-c.-difficile-infection" class="slide level2">
<h2>The human gut microbiome changes during <em>C. difficile</em> infection</h2>

<img data-src="img/cdiff-infection.png" class="r-stretch"><div class="footer">
<p>Created with <a href="biorender.com" data-fig-align="center">BioRender</a></p>
</div>
<aside class="notes">
<ul>
<li><strong>CDI</strong> - treatment with antibiotics alters the taxonomic composition of the gut microbiome, allowing <em>C. diff</em> to colonize the gut.</li>
<li>mouse studies have found differences resistance to colonization, time to clearance, and severe outcomes based on the initial state of the gut microbiome.</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="how-to-study-the-gut-microbiome" class="slide level2">
<h2>How to study the gut microbiome</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/microbiome-research.png" height="600"></p>
</figure>
</div>
<div class="footer">
<p>Created with <a href="biorender.com">BioRender</a></p>
</div>
<aside class="notes">
<p>analyze microbial sequences to determine which microbes are there and how abundant they are.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="machine-learning-for-science-health-care" class="slide level2">
<h2>Machine learning for science &amp; health care</h2>
<p>Supervised machine learning (ML) - computational techniques that identify patterns in large datasets to classify samples or predict outcomes.</p>
<h5 id="applications">Applications:</h5>
<ul>
<li>Diagnose colorectal cancer early and less invasively than colonoscopy.</li>
<li>Identify COVID-19 patients at risk of clinical deterioration.</li>
<li>Predict severe outcomes of <em>C. difficile</em> infections.</li>
</ul>
<aside class="notes">
<p>Traditional microbial ecology methods aren’t sufficient to distinguish between disease states. individual microbes aren’t enough to explain it; something more complex is going on</p>
<ul>
<li>high dimensionality: 20k OTUs</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="overview" class="slide level2">
<h2>Overview</h2>
<div class="column" style="width:72%;">
<ol type="1">
<li>Improve methods for processing microbiome data.</li>
<li>Predict severe <em>C. difficile</em> infections from gut microbiome composition.</li>
<li>Contribute to democratizing data science.</li>
</ol>
</div>
</section>
<section id="how-to-characterize-microbiomes" class="slide level2">
<h2>How to characterize microbiomes</h2>
<p><br></p>

<img data-src="img/amplicon-vs-metagenomics_edited.png" class="r-stretch quarto-figure-center"><div class="footer">
<p>Adapted from: Lee (2019). <a href="https://astrobiomike.github.io/misc/amplicon_and_metagen">JOSE</a></p>
</div>
<aside class="notes">
<p>to characterize the bacteria in a microbial community, the gene encoding the 16S ribosomal RNA subunit is commonly chosen as the target gene to amplify and sequence. after sequencing, the goal is to identify which bacteria are present and how abundant are they in each sample. then we can compare across samples to identify patterns that differentiate disease states or outcomes. ideally, for each sequence we would identify which species it came from, but that’s not possible with amplicon sequencing.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="difficulties-in-bacterial-taxonomy" class="slide level2">
<h2>Difficulties in bacterial taxonomy</h2>
<div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/biological-taxonomy.png"></p>
</figure>
</div>
</div>
<div class="column smaller" style="width:45%;">
<ul>
<li>Inconsistent species concept for microbes.</li>
<li>Most microbes are not easy to culture, thus are not well-characterized.</li>
<li>Many bacteria have multiple copies of target genes for amplicon sequencing.</li>
</ul>
</div>
<aside class="notes">
<p>why OTUs?</p>
<ul>
<li>Inconsistent species concept for bacteria.
<ul>
<li>Binary fission - bacteria reproduce by cloning themselves.</li>
<li>Horizontal gene transfer - bacteria can swap genes with each other.</li>
<li>Their phenotypic properties do not reveal common ancestry.</li>
</ul></li>
<li>Amplicon sequences do not contain enough information to identify bacteria at the species level.
<ul>
<li>16S highly conserved; 16S rRNA trees informed classifications at and above the rank of genus (https://www.nature.com/articles/s41396-021-00941-x).</li>
</ul></li>
<li>Many bacteria have multiple copies of the 16S rRNA gene, the most popular marker gene for amplicon sequencing bacterial communities.</li>
<li>Using every sequence (ASV level) as a feature would split them into separate taxonomic groups.</li>
<li>Bacteria that are difficult to culture are not well-characterized.
<ul>
<li>So we have names assigned to many bacteria that grow easily in the lab, but we’re missing information for lots of bacteria that are hard to culture.</li>
<li>This leaves holes in public reference databases.</li>
<li>Database-independent clustering methods allow exploring microbial “dark matter”.</li>
</ul></li>
</ul>
<p>Solution: cluster the amplicon sequences into operational taxonomic units (OTU) based on how similar the sequences are to each other. OTUs can generally be identified down to the genus level.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<div class="footer">
<p>Created with <a href="biorender.com">BioRender</a></p>
</div>
</section>
<section id="clustering-amplicon-sequences-into-operational-taxonomic-units-otus" class="slide level2" data-background-image="img/logo/mothur_RGB.png" data-background-size="120px" data-background-position="top 20px right 20px">
<h2>Clustering amplicon sequences into Operational Taxonomic Units (OTUs)</h2>
<h4 id="de-novo-opticlust-algorithm"><em>De novo</em> OptiClust algorithm</h4>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/de-novo-clustering-0.png" height="500"></p>
</figure>
</div>
<div class="footer">
<p>Created with <a href="biorender.com">BioRender</a></p>
</div>
</section>
<section id="clustering-amplicon-sequences-into-operational-taxonomic-units-otus-1" class="slide level2" data-background-image="img/logo/mothur_RGB.png" data-background-size="120px" data-background-position="top 20px right 20px">
<h2>Clustering amplicon sequences into Operational Taxonomic Units (OTUs)</h2>
<h4 id="de-novo-opticlust-algorithm-1"><em>De novo</em> OptiClust algorithm</h4>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/de-novo-clustering-1.png" height="500"></p>
</figure>
</div>
<div class="footer">
<p>Created with <a href="biorender.com">BioRender</a></p>
</div>
<div class="section-logo">
<p><img data-src="img/hex/logo_mothur.png"></p>
</div>
</section>
<section id="clustering-amplicon-sequences-into-operational-taxonomic-units-otus-2" class="slide level2" data-background-image="img/logo/mothur_RGB.png" data-background-size="120px" data-background-position="top 20px right 20px">
<h2>Clustering amplicon sequences into Operational Taxonomic Units (OTUs)</h2>
<h4 id="de-novo-opticlust-algorithm-2"><em>De novo</em> OptiClust algorithm</h4>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/de-novo-clustering-2.png" height="500"></p>
</figure>
</div>
<div class="footer">
<p>Created with <a href="biorender.com">BioRender</a></p>
</div>
</section>
<section id="clustering-amplicon-sequences-into-operational-taxonomic-units-otus-3" class="slide level2" data-background-image="img/logo/mothur_RGB.png" data-background-size="120px" data-background-position="top 20px right 20px">
<h2>Clustering amplicon sequences into Operational Taxonomic Units (OTUs)</h2>
<h4 id="de-novo-opticlust-algorithm-3"><em>De novo</em> OptiClust algorithm</h4>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/de-novo-clustering-3.png" height="500"></p>
</figure>
</div>
<div class="footer">
<p>Created with <a href="biorender.com">BioRender</a></p>
</div>
<aside class="notes">
<p>TODO highlight limitations here</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="clustering-amplicon-sequences-into-operational-taxonomic-units-otus-4" class="slide level2" data-background-image="img/logo/mothur_RGB.png" data-background-size="120px" data-background-position="top 20px right 20px">
<h2>Clustering amplicon sequences into Operational Taxonomic Units (OTUs)</h2>
<h4 id="reference-based-algorithm">Reference-based algorithm</h4>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/reference-clustering-1.png" height="500"></p>
</figure>
</div>
<div class="footer">
<p>Created with <a href="biorender.com">BioRender</a></p>
</div>
</section>
<section id="clustering-amplicon-sequences-into-operational-taxonomic-units-otus-5" class="slide level2" data-background-image="img/logo/mothur_RGB.png" data-background-size="120px" data-background-position="top 20px right 20px">
<h2>Clustering amplicon sequences into Operational Taxonomic Units (OTUs)</h2>
<h4 id="reference-based-algorithm-1">Reference-based algorithm</h4>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/reference-clustering-2.png" height="500"></p>
</figure>
</div>
<div class="footer">
<p>Created with <a href="biorender.com">BioRender</a></p>
</div>
</section>
<section id="clustering-amplicon-sequences-into-operational-taxonomic-units-otus-6" class="slide level2" data-background-image="img/logo/mothur_RGB.png" data-background-size="120px" data-background-position="top 20px right 20px">
<h2>Clustering amplicon sequences into Operational Taxonomic Units (OTUs)</h2>
<h4 id="reference-based-algorithm-2">Reference-based algorithm</h4>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/reference-clustering-3.png" height="500"></p>
</figure>
</div>
<div class="footer">
<p>Created with <a href="biorender.com">BioRender</a></p>
</div>
<aside class="notes">
<p>Limitations of existing clustering methods</p>
<ul>
<li><em>De novo</em> clustering results in inconsistent OTUs when adding new data.</li>
<li>Reference-based clustering results in lower quality OTUs than <em>de novo</em>.</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="optifit" class="slide level2" data-background-image="img/logo/mothur_RGB.png" data-background-size="120px" data-background-position="top 20px right 20px">
<h2>OptiFit</h2>
<h3 id="reference-based-clustering-to-de-novo-otus">reference-based clustering to <em>de novo</em> OTUs</h3>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/optifit-clustering-1.png" height="500"></p>
</figure>
</div>
<div class="footer">
<p>Created with <a href="biorender.com">BioRender</a></p>
</div>
</section>
<section id="optifit-1" class="slide level2" data-background-image="img/logo/mothur_RGB.png" data-background-size="120px" data-background-position="top 20px right 20px">
<h2>OptiFit</h2>
<h3 id="reference-based-clustering-to-de-novo-otus-1">reference-based clustering to <em>de novo</em> OTUs</h3>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/optifit-clustering-2.png" height="500"></p>
</figure>
</div>
<div class="footer">
<p>Created with <a href="biorender.com">BioRender</a></p>
</div>
</section>
<section id="optifit-2" class="slide level2" data-background-image="img/logo/mothur_RGB.png" data-background-size="120px" data-background-position="top 20px right 20px">
<h2>OptiFit</h2>
<h3 id="reference-based-clustering-to-de-novo-otus-2">reference-based clustering to <em>de novo</em> OTUs</h3>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/optifit-clustering-3.png" height="500"></p>
</figure>
</div>
<div class="footer">
<p>Created with <a href="biorender.com">BioRender</a></p>
</div>
</section>
<section id="optifit-benchmarking-results" class="slide level2" data-background-image="img/logo/mothur_RGB.png" data-background-size="120px" data-background-position="top 20px right 20px">
<h2>OptiFit benchmarking results</h2>
<div class="cell" data-hash="presentation_cache/revealjs/mcc_36cb0c10ec8c7db76bd2d8b1cf737997">
<div class="cell-output-display">
<p><img data-src="figures/mcc-1.png" width="1000" height="450"></p>
</div>
</div>
<div class="footer">
<p>Sovacool <em>et al.</em> (2022). <a href="https://doi.org/10.1128/msphere.00916-21">mSphere</a></p>
</div>
<aside class="notes">
<p>so we benchmarked the optifit algorithm alongside a few other clustering strategies. We tested these algorithms with four public datasets from human gut, mouse gut, soil, and marine microbiome samples. For the sake of time I’m only showing the results from the human dataset here.</p>
<p>we used the Matthews Correlation Coefficient (MCC) to measure how well each algorithm adhered to the distance threshold. The MCC ranges from -1 to 1, with 1 being perfect quality where all pairs of sequences that are within the distance threshold are in the same OTU and all pairs of sequences that are more dissimilar than the threshold are in different OTUs.</p>
<p>on the y-axis we have each clustering strategy. first is <em>de novo</em> clustering with OptiClust and <em>de novo</em> clustering with another program called VSEARCH. we then tested OptiFit with a self-split strategy, where we clustered half of the sequences <em>de novo</em> then fit the remaining sequences to the <em>de novo</em> OTUs as a reference. the self-split strategy with OptiFit performed just as well as OptiClust in terms of MCC. Despite issues with public databases, traditional reference-based clustering to an external database is still fairly common, so we tested OptiFit’s ability to fit sequences to three databases: greengenes, silva, and the RDP. for database-depending clustering was highly dependent on the size of the database.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="optifit-benchmarking-results-1" class="slide level2" data-background-image="img/logo/mothur_RGB.png" data-background-size="120px" data-background-position="top 20px right 20px">
<h2>OptiFit benchmarking results</h2>
<div class="cell" data-hash="presentation_cache/revealjs/sec_b182ad83e7e7f83f5fbed767c9650c29">
<div class="cell-output-display">
<p><img data-src="figures/sec-1.png" width="1000" height="450"></p>
</div>
</div>
<div class="footer">
<p>Sovacool <em>et al.</em> (2022). <a href="https://doi.org/10.1128/msphere.00916-21">mSphere</a></p>
</div>
</section>
<section id="optifit-improves-reference-based-clustering" class="slide level2" data-background-image="img/logo/mothur_RGB.png" data-background-size="120px" data-background-position="top 20px right 20px">
<h2>OptiFit improves reference-based clustering</h2>
<ul>
<li>New method to use <em>de novo</em> OTUs as a reference for new sequences.</li>
<li>Clusters sequences at the same quality as <em>de novo</em> methods.</li>
<li>Outperforms other reference-based methods in terms of quality.
<ul>
<li>Trade-off: longer run time.</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Follow-up: OptiFit is suitable for machine learning.</li>
</ul>
</div>
<div class="footer">
<p>Armour <em>et al.</em> (2023). <a href="10.1101/2022.09.01.506299">bioRxiv</a> <br> Sovacool <em>et al.</em> (2022). <a href="https://doi.org/10.1128/msphere.00916-21">mSphere</a></p>
</div>
<aside class="notes">
<p>So in conclusion for this project, we found that OptiFit is a new method that improves reference-based clustering.</p>
<p>In a follow-up study, Courtney Armour demonstrated that you can use OptiFit to fit new data to existing OTUs and machine learning models tested with the new data performed just as well as if all of the data had been clustered <em>de novo</em>.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="overview-1" class="slide level2">
<h2>Overview</h2>
<div class="column" style="width:72%;">
<ol type="1">
<li>Improve methods for processing microbiome data.</li>
<li><strong>Predict severe <em>C. difficile</em> infections from gut microbiome composition.</strong></li>
<li>Contribute to democratizing data science.</li>
</ol>
</div>
</section>
<section id="c.-difficile-infection-cdi" class="slide level2">
<h2><em>C. difficile</em> infection (CDI)</h2>
<p><img data-src="img/Cdiff-Factsheet-P_impact-risk.png" height="500"></p>
<div class="footer">
<p><a href="https://www.cdc.gov/cdiff/">cdc.gov/cdiff</a></p>
</div>
<aside class="notes">
<p><em>Clostridioides difficile</em> is a bacterium that causes infection.</p>
<p>CDI has a mortality rate of about 8% and that risk is higher in older patients. CDI is a significant burden on the US health care system.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="clinical-data-predict-complicated-cdi" class="slide level2">
<h2>Clinical data predict complicated CDI</h2>
<div class="column" style="width:60%;">
<p><img data-src="img/li_ehr_auroc.jpeg"></p>
<div class="smaller-text">
<p>median AUROC: <strong>0.69</strong></p>
</div>
</div>
<div class="column smaller-text" style="width:35%;">
<ul>
<li>Motivation: inform clinicians on which CDI patients may be most at risk of severe outcomes in order to tailor treatments.</li>
<li>Electronic health record (EHR) data used as features to predict whether disease-related complications occurred.</li>
</ul>
</div>
<div class="footer">
<p>Li <em>et al.</em> (2019). <a href="https://doi.org/10.1093/ofid/ofz186">OFID</a></p>
</div>
<aside class="notes">
<p>Area under the receiver-operator characteristic curve (AUROC) - given a randomly selected severe case and a not severe case, what is the probability that the model correctly ranks the severe case as having a higher risk.</p>
<p>CDI treatment. Li et al.&nbsp;(2019). https://academic.oup.com/ofid/article/6/5/ofz186/5475497 “The development and validation of EHR-based risk stratification models for predicting complicated CDI could eventually help clinicians tailor treatments to individuals. On the day of CDI diagnosis, a patient’s estimated risk for complications could serve as an adjunct, easily obtainable resource for clinical decision support. Treatment decisions such as whether to use high-dose vancomycin or perform a loop ileostomy with antegrade vancomycin infusions [23] often do not occur until complicated CDI has already set in. In severe cases, early aggressive therapy can positively impact the course. However, invasive treatments such as enemas (fecal microbiota transplantation or vancomycin) and surgery are optimally used in only select patients, and such decisions lack the rigorous guidelines associated with initial treatment.”</p>
<p>Would help clinicians tailor treatment options early on. Whether to go with an aggressive treatment plan or not.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="section" class="slide level2">
<h2></h2>
<p><br></p>
<div class="big-middle">
<p>Can we use the composition of the gut microbiome to predict CDI severity?</p>
</div>
<aside class="notes">
<p>Clinical factors have been used to predict CDI severity, and the initial state of the gut microbiome plays a role in colonization resistance and clearance.</p>
<p>Our central question is: Can we use the composition of the gut microbiome to predict CDI severity?</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="how-to-define-cdi-severity" class="slide level2 smaller">
<h2>How to define CDI severity</h2>
<p><br></p>
<p><img data-src="projects/severe-cdi/figures/severity_flowchart.png" height="600"></p>
<div class="absolute narrow-frame fragment" style="top: 70px; left: 180px; width: 35%; ">
<p>1,257 CDI patient stool samples collected on the day of diagnosis</p>
</div>
<div class="fragment filled-box absolute" style="top: 300px; left: 850px; ">
<div class="cell" data-hash="presentation_cache/revealjs/sample_counts_3db2dbf20cec2c8615b80d3ef4bc03f5">
<div class="cell-output-display">
<table class="table" data-quarto-postprocess="true" style="font-size: 20px; margin-left: auto; margin-right: auto;">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Severity</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">n</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">% severe</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">IDSA</td>
<td style="text-align: right;">1,072</td>
<td style="text-align: right;">34.2</td>
</tr>
<tr class="even">
<td style="text-align: left;">All-cause</td>
<td style="text-align: right;">1,218</td>
<td style="text-align: right;">7.1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Attributable</td>
<td style="text-align: right;">1,178</td>
<td style="text-align: right;">2.2</td>
</tr>
</tbody>
</table>


</div>
</div>
</div>
<div class="fragment filled-box absolute" style="top: 300px; left: 850px; ">
<div class="cell" data-hash="presentation_cache/revealjs/sample_counts_pragmatic_b6a631c85a9b70d0f6f8518ee6a17ab4">
<div class="cell-output-display">
<table class="table" data-quarto-postprocess="true" style="font-size: 20px; margin-left: auto; margin-right: auto;">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Severity</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">n</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">% severe</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">IDSA</td>
<td style="text-align: right;">1,072</td>
<td style="text-align: right;">34.2</td>
</tr>
<tr class="even">
<td style="text-align: left;">All-cause</td>
<td style="text-align: right;">1,218</td>
<td style="text-align: right;">7.1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Attributable</td>
<td style="text-align: right;">1,178</td>
<td style="text-align: right;">2.2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Pragmatic</td>
<td style="text-align: right;">1,218</td>
<td style="text-align: right;">5.4</td>
</tr>
</tbody>
</table>


</div>
</div>
</div>
<aside class="notes">
<p>First we need to talk about how to define CDI severity. There are a number of ways to do it. Following this branch to the left, you can consider serum biomarkers like creatinine level and white blood cell count. The Infectious Diseases Society of America uses elevated creatinine and wbc count as a measure of CDI severity. It effectively tells clinicians “how sick is this patient right now”. These data are also relatively easy to obtain.</p>
<p>On the other hand, we can look at the ultimate outcomes of the infection at 30 days since diagnosis.</p>
<p>So for each patient sample collected on the day of diagnosis, we performed amplicon sequencing and clustered sequences into operational taxonomic units. We used the relative abundances of our OTUs as features for training machine learning models to predict severe CDI outcomes.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="training-machine-learning-models" class="slide level2">
<h2>Training machine learning models</h2>

<img data-src="img/topcuoglu_framework.jpeg" class="r-stretch"><div class="footer">
<p>Topçuoğlu <em>et al.</em> (2020). <a href="https://doi.org/10.1128/mBio.00434-20">mBio</a></p>
</div>
</section>
<section id="model-performance" class="slide level2 smaller">
<h2>Model performance</h2>
<p>Random Forests trained on 100x train/test splits of each dataset</p>

<img data-src="figures/roc_curves-1.png" style="width:100.0%" class="r-stretch"><aside class="notes">
<p>ROC = trade-off between correctly predicting positive vs negative samples. The baseline is 0.5 for a model that performs no better than random.</p>
<p>sensitivity/recall = <span class="math inline">\(TPR = \frac{TP}{TP + FN}\)</span></p>
<p>Out of all positive samples, how many correctly predicted positive.</p>
<p>specificity = <span class="math inline">\(TNR = \frac{TN}{TN + FP} = 1 - FPR\)</span></p>
<p>Out of all negative samples, how many correctly predicted negative.</p>
<p><span class="math inline">\(FPR = \frac{FP}{FP + TN}\)</span></p>
<p>Out of all negative samples, how many incorrectly predicted positive.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="model-performance-1" class="slide level2 smaller">
<h2>Model performance</h2>
<p>Random Forests trained on 100x train/test splits of each dataset</p>

<img data-src="figures/auroc-1.png" width="960" class="r-stretch"><div class="absolute medium-text frame-box fragment" style="top: 20px; right: 100px; width: 250px; height: 80px; ">
<p>EHR-based models from Li <em>et al.</em></p>
<p>Median AUROC: <strong>0.69</strong></p>
</div>
<aside class="notes">
<p>We can also summarize ROC curves with the area under the roc curve (AUROC).</p>
<p>ROC = trade-off between correctly predicting positive vs negative samples.</p>
<p>sensitivity/recall = <span class="math inline">\(TPR = \frac{TP}{TP + FN}\)</span></p>
<p>Out of all positive samples, how many correctly predicted positive.</p>
<p>specificity = <span class="math inline">\(TNR = \frac{TN}{TN + FP} = 1 - FPR\)</span></p>
<p>Out of all negative samples, how many correctly predicted negative.</p>
<p><span class="math inline">\(FPR = \frac{FP}{FP + TN}\)</span></p>
<p>Out of all negative samples, how many incorrectly predicted positive.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="model-performance-on-severe-cases" class="slide level2 smaller">
<h2>Model performance on severe cases</h2>
<p>Random Forests trained on 100x train/test splits of each dataset</p>

<img data-src="figures/prc-1.png" width="960" class="r-stretch"><aside class="notes">
<p>To ensure the AUROC isn’t artificially inflated due to the high proportion of negative samples in our dataset, we also plotted precision-recall curves.</p>
<p>ROC = trade-off between correctly predicting positive vs negative samples.</p>
<p>precision = <span class="math inline">\(PPV = \frac{TP}{TP + FP}\)</span></p>
<p>Out of the samples that were predicted positive, how many were correctly predicted.</p>
<p>sensitivity/recall = <span class="math inline">\(TPR = \frac{TP}{TP + FN}\)</span></p>
<p>Out of all positive samples, how many correctly predicted.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="top-otus-contributing-to-model-performance" class="slide level2">
<h2>Top OTUs contributing to model performance</h2>

<img data-src="projects/severe-cdi/figures/feature-importance_full.png" class="r-stretch"><aside class="notes">
<p>Each of the most important OTUs were more abundant in severe cases, and each have been associated with loss of colonization resistance or longer time to clearance in mouse studies of CDI. In particular, Enterococcus has been found to be associated with increased risk of severe outcomes in mouse studies.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="clinical-value-of-prediction-models" class="slide level2">
<h2>Clinical value of prediction models</h2>
<ul>
<li>The pragmatic severity model performed just as well as an EHR-based model.</li>
<li>The most important OTUs for model performance concord with prior studies.</li>
<li>Open question: would deploying EHR or OTU-based models improve clinical outcomes?</li>
</ul>
<aside class="notes">
<ul>
<li>The pragmatic severity model performed just as well as an EHR-based model.</li>
<li>The most important OTUs for model performance concord with prior studies.</li>
<li>However, decent model performance is not enough to justify model deployment.</li>
<li>Open question: would deploying EHR or OTU-based models improve clinical outcomes?
<ul>
<li>A better comparison would extract EHRs from this same dataset and compare performance for models trained on EHRs, OTUs, or both.</li>
<li>Clinical value of prediction models is also highly dependent on treatment options available to help patients identified as high risk.</li>
<li>To assess clinical value, we need to consider the trade-offs of the resources required to train and deploy these models versus the potential benefit gained for patients.</li>
</ul></li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="overview-2" class="slide level2">
<h2>Overview</h2>
<div class="column" style="width:72%;">
<ol type="1">
<li>Improve methods for processing microbiome data.</li>
<li>Predict severe <em>C. difficile</em> infections from gut microbiome composition.</li>
<li><strong>Contribute to democratizing data science.</strong></li>
</ol>
</div>
</section>
<section id="democratizing-data-science" class="slide level2">
<h2>Democratizing data science</h2>
<ul>
<li>Make data science tools more accessible to researchers from non-computational backgrounds.</li>
<li>Disseminate user-friendly tools &amp; curricula with open source licenses.</li>
<li>Promote diversity, equity, and inclusion in data science.</li>
</ul>
<aside class="notes">
<p>To these ends, we developed resources targeting 3 key audiences. high schoolers, academic researchers, novice ML practitioners</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="addressing-the-gender-gap-in-stem" class="slide level2">
<h2>Addressing the gender gap in STEM</h2>
<ul>
<li>18% of Computer Science degrees are awarded to women.</li>
<li>U-M Bioinformatics PhD graduates: 25% women as of 2021.</li>
<li>Most of us (women in U-M Bioinformatics) didn’t learn to code until college or later.</li>
</ul>
<div class="fragment">
<p><strong>Solution:</strong> start a Girls Who Code club focused on teaching coding for data science to high school students.</p>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/logo/gwc.png" width="475"></p>
</figure>
</div>
</div>
</section>
<section id="intro-to-python-for-data-science" class="slide level2" data-background-image="img/logo/gwc.png" data-background-size="250px" data-background-position="top 20px right 20px">
<h2>Intro to Python for Data Science</h2>
<h3 id="for-girls-who-code-at-u-m-dcmb">for Girls Who Code at U-M DCMB</h3>

<img data-src="projects/girls-who-code/paper/lesson-modules.png" class="r-stretch"><div class="footer">
<p>Duda<sup>*</sup> and Sovacool<sup>*</sup> <em>et al.</em> (2021). <a href="https://doi.org/10.21105/jose.00138">JOSE</a></p>
</div>
<aside class="notes">
<p>teach high schoolers the basics of coding in Python for data science. from knowing no programming to being able to analyze a data science and present findings at the end of the club.</p>
<p>live coding. hands on lessons and practice notebooks. pivot to flipped format for pandemic.</p>
<ol type="1">
<li>Each lesson begins with a recapping of the relevant core skills presented in the previous lessons.</li>
<li>All lessons are designed to be taught via 15-minute live-coding sessions. This method is used by The Carpentries and is demonstrated to be an effective method that engages learners (Nederbragt et al., 2020; Wilson, 2016) since learners must actively engage with the material and deal with errors and bugs as they arise.</li>
<li>Each lesson ends with a summary of core skills presented within the material.</li>
<li>Each short lesson is also accompanied by a subsequent 10-minute independent practice, providing further opportunity for practical experience implementing the coding skill at hand and testing learners’ understanding of the content.</li>
</ol>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="impact-of-girls-who-code" class="slide level2" data-background-image="img/logo/gwc.png" data-background-size="250px" data-background-position="top 20px right 20px">
<h2>Impact of Girls Who Code</h2>
<ul>
<li>135+ graduates of the club &amp; summer camp.</li>
<li>Graduates improve skills in coding, problem solving, collaboration, and self-confidence.</li>
</ul>
<blockquote>
<p>Because of GWC, I learned about bioinformatics. I was very interested in it, because it combined my interests in programming and research… I’m pretty certain I want to go into bioinformatics now.</p>
</blockquote>
<p><br></p>
<blockquote>
<p>I plan to go to college for Computer Science and get a robotics minor when my college offers it. GWC has inspired me to consider pursuing a Masters or PhD in CS as well as take some electives in Data Science.</p>
</blockquote>
</section>
<section id="coding-for-reproducible-research" class="slide level2" data-background-image="img/logo/software-carpentry.png" data-background-size="200px" data-background-position="top 30px right 20px">
<h2>Coding for Reproducible Research</h2>
<ul>
<li>34,000+ learners have attended Software Carpentry workshops worldwide since 2010.</li>
<li>U-M instance of Software Carpentry active since 2016.</li>
</ul>
<div class="fragment">
<p><img data-src="img/swc-lessons.png"></p>
</div>
<div class="footer">
<p>https://software-carpentry.org/lessons/</p>
</div>
<aside class="notes">
<ul>
<li>Software Carpentry is an international non-profit that teaches grad students, postdocs, and other researchers how to perform reproducible computational science via interactive coding workshops.</li>
<li>Over 34,000 learners have attended since 2010, ranging from biologists to physicists to engineers and economists.</li>
<li>U-M instance of Software Carpentry active since 2016.</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="integrated-workshop-curriculum" class="slide level2" data-background-image="img/logo/software-carpentry.png" data-background-size="200px" data-background-position="top 30px right 20px">
<h2>Integrated workshop curriculum</h2>
<h3 id="teaching-r-the-unix-shell-and-git">teaching R, the Unix shell, and Git</h3>

<img data-src="projects/software-carpentry/paper/curriculum-overview.png" class="r-stretch"><div class="footer">
<p>Lapp<sup>*</sup> and Sovacool<sup>*</sup> <em>et al.</em> (2022). <a href="https://doi.org/10.21105/jose.00144">JOSE</a></p>
</div>
</section>
<section id="impact-of-software-carpentry" class="slide level2" data-background-image="img/logo/software-carpentry.png" data-background-size="200px" data-background-position="top 30px right 20px">
<h2>Impact of Software Carpentry</h2>
<h3 id="pilot-workshop-pre-post-survey">Pilot workshop pre &amp; post survey</h3>

<img data-src="projects/software-carpentry/paper/survey-results.png" class="r-stretch"><div class="footer">
<p>Lapp<sup>*</sup> and Sovacool<sup>*</sup> <em>et al.</em> (2022). <a href="https://doi.org/10.21105/jose.00144">JOSE</a></p>
</div>
<aside class="notes">
<p>New curriculum used in 7 workshops to date.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="reproducible-ml-pipelines-with-mikropml" class="slide level2" data-background-image="img/hex/logo_mikropml.png" data-background-size="150px" data-background-position="top 20px right 20px">
<h2>Reproducible ML pipelines with mikropml</h2>
<div class="absolute" style="top: 50px; right: 100px; ">
<p><code>"meek-rope em el"</code></p>
</div>
<p><br></p>
<div class="column" style="width:80%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/mikropml-pipeline_crop.png"></p>
</figure>
</div>
</div>
<div class="column fragment" style="width:19%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/mikropml-snakemake.png"></p>
</figure>
</div>
</div>
<div class="footer">
<p>Topçuoğlu<sup>*</sup>, Lapp<sup>*</sup>, Sovacool<sup>*</sup> <em>et al.</em> (2021). <a href="https://doi.org/10.21105/joss.03073">JOSS</a></p>
</div>
<aside class="notes">
<ul>
<li>Typical microbiologist lacks ML expertise</li>
<li>Many microbiologists have enough computational skills to be “dangerous”</li>
<li>Common pitfalls
<ul>
<li>Lack of transparency in method implementation</li>
<li>No separate held-out test data</li>
<li>Not reporting variation in the predictive performance on different folds of cross-validation, or cross-validation vs.&nbsp;testing performances</li>
</ul></li>
</ul>
<p>Goals</p>
<ul>
<li>Restructure our ML framework to improve reusability, scalability, &amp; reproducibility.</li>
<li>Strike a balance between easy-to-use interface for beginners with reasonable default options, with enough flexibility for advanced users to customize options.</li>
<li>We implemented this as an R package and also included a snakemake workflow demonstrating how to use the package in a high performance computing environment.
<ul>
<li>We intend to use it alongside other software in the lab: mothur for processing amplicon sequences and schtools – schloss lab tools for reproducible microbiome research.</li>
<li>But mikropml is applicable outside just the microbiome field. So we published it in the Journal of Open Source Software</li>
</ul></li>
</ul>
<p>best practices in swe</p>
<ul>
<li>GitFlow, issue tracking, peer review</li>
<li>Unit tests + CI</li>
<li>Document functions &amp; vignettes</li>
<li>Semantic versioning, changelog</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="mikropml-impact" class="slide level2 smaller" data-background-image="img/hex/logo_mikropml.png" data-background-size="150px" data-background-position="top 20px right 20px">
<h2>mikropml impact</h2>
<script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
<script async="" src="https://badge.dimensions.ai/badge.js" charset="utf-8"></script>
<div class="column" style="width:70%;">
<p><img data-src="projects/pkg-downloads/figures/plot-downloads-time_no-badge-1.png" height="550"></p>
</div>
<div class="column" style="width:15%;">
<p><br> <br></p>
<p>CRAN: <img src="https://cranlogs.r-pkg.org/badges/grand-total/mikropml"></p>
<p>conda-forge: <img data-src="https://anaconda.org/conda-forge/r-mikropml/badges/downloads.svg"></p>
<p><span class="__dimensions_badge_embed__" data-doi="10.21105/joss.03073" data-hide-zero-citations="true" data-style="large_rectangle" data-legend="hover-right"></span></p>
</div>
<aside class="notes">
<div class="column" style="width:48%;">
<p><strong>In the lab</strong></p>
<ul>
<li>CRC classification across taxonimic levels</li>
<li>Mouse <em>C. diff</em> colonization &amp; clearance</li>
<li>Mouse models of colitis</li>
</ul>
</div>
<div class="column" style="width:48%;">
<p><strong>In the wild</strong></p>
<ul>
<li>Diet and antimicrobial resistance</li>
<li>Evolution of grasses</li>
<li>Psychology research</li>
<li>Political science</li>
</ul>
</div>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="summary" class="slide level2">
<h2>Summary</h2>
<ul>
<li>Improved OTU clustering algorithms to enable machine learning for microbiome research.</li>
<li>Applied machine learning to predict severe <em>C. difficile</em> infections, demonstrating similar performance as models trained on Electronic Health Records.</li>
<li>Contributed to democratizing data science for three key audiences.</li>
</ul>
<div class="column smaller-text fragment" style="width:49%;">
<!-- -->
<h5 id="post-phd-plans">post-PhD plans</h5>
<p>Bioinformatics Software Engineer for Frederick National Lab (NIH/NCI)</p>
<p><img data-src="img/logo/fnlcr.png"></p>
</div>
<aside class="notes">
<ul>
<li>Improved OTU clustering algorithms to enable ML for microbiome research.</li>
<li>Applied ML to predict severe CDIs, demonstrating similar performance as EHR models.</li>
<li>Contributed to democratizing data science for three key audiences: high schoolers, researchers, and novice ML practitioners.</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="acknowledgements" class="slide level2">
<h2>Acknowledgements</h2>
<div class="column large-text" style="width:30%;">
<p><strong>Schloss Lab</strong> <br> Patrick Schloss <br> Megan Coden <br> Sarah Lucas <br> Courtney Armour <br> Allison Mason <br> Adena Collens</p>
<p><strong>alumni</strong> <br> Nick Lesniak <br> Sarah Tomkovich <br> Sarah Westcott <br> Katie McBride <br> Jay Moltzau <br> Begüm Topçuoğlu <br> Joshua Stough <br> Will Close <br> Ada Hagan <br> Kaitlin Flynn <br></p>
</div>
<div class="column large-text" style="width:30%;">
<p><strong>Curricula contributors</strong> <br> Marlena Duda <br> Zena Lapp <br> Brooke Wolford <br> Negar Farzaneh <br> Vy Nguyen <br> Sarah Haynes <br> Hayley Falk <br> Katie Furman <br> Logan Walker <br> Rucheng Diao <br> Morgan Oneka <br> Audrey Drotos <br> Gabrielle Dotson <br> April Kriebel <br> Lucy Meng <br> Stephanie Theide <br> Dana King <br> Catherine Barnier <br> Matthew Flickinger <br> Jule Krüger <br> Maya Lapp <br> Jason Tallant <br></p>
</div>
<div class="column large-text" style="width:30%;">
<p><strong>Collaborators</strong> <br> <strong>&amp; Committee</strong> <br> Krishna Rao <br> Vince Young <br> Jenna Wiens <br> Greg Dick <br></p>
<p><strong>Funding</strong> <br> NIH T32 GM070449 <br> NIH U01AI124255 <br></p>
</div>
</section>
<section id="section-1" class="slide level2">
<h2></h2>
<div class="column" style="width:70%;">
<p><img data-src="projects/best-tally/figures/plot_blank-1.png"></p>
</div>
<aside class="notes">
<p>I have one more plot to show you all. Early on in my PhD, my friends and I noticed that there was a particular phrase we kept saying fairly regularly. And not just us, but also colleagues and collaborators and others in our scientific circles.</p>
<p>So I decided to keep a tally.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="section-2" class="slide level2">
<h2></h2>
<div class="column" style="width:70%;">
<p><img data-src="projects/best-tally/figures/plot_line-1.png"></p>
</div>
<div class="column large-text fragment" style="width:25%;">
<p><br></p>
<h4 id="phrases-included">Phrases included</h4>
<ul>
<li>“Pat’s the best”</li>
<li>“Pat is a <em>delight</em>.”</li>
<li>“Wow this is incredible. Pat is literally a hero!”</li>
<li>“Your PI is amazing”</li>
</ul>
</div>
<aside class="notes">
<p>That phrase is “Pat is the best”.</p>
<p>For the sake of completeness I also included similar phrases uttered such as “Pat is a delight”, “Pat is literally a hero”, and “Your PI is amazing”.</p>
<p>There were <strong>so</strong> many times throughout my PhD when Pat did or said something that elicited this response. From his response to COVID, to his support of my work with Girls Who Code and other volunteer efforts, all the ups and downs of grad school… Pat even made an educational YouTube channel to teach people how to code.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="section-3" class="slide level2">
<h2></h2>
<div class="column" style="width:70%;">
<p><img data-src="projects/best-tally/figures/plot_line_dry-1.png"></p>
</div>
<div class="column large-text" style="width:25%;">
<p><br></p>
<h4 id="phrases-included-1">Phrases included</h4>
<ul>
<li>“Pat’s the best”</li>
<li>“Pat is a <em>delight</em>.”</li>
<li>“Wow this is incredible. Pat is literally a hero!”</li>
<li>“Your PI is amazing”</li>
</ul>
</div>
<aside class="notes">
<p>(This one is my favorite thumbnail from his videos.)</p>
<p>Finally, Pat has persistently done a great job of assembling a lab.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="section-4" class="slide level2" data-background-image="img/acknowledgements-schlabbies.jpg" data-background-size="contain">
<h2></h2>
<aside class="notes">
<p>I thanked members of the Schloss Lab earlier for your scientific contributions but now I’m thanking you again for being wonderful people because you are all just that special and played such a huge role in shaping me as a scientist. Many lab mates have come and gone over the years and every one I’ve interacted with has individually contributed to an awesome positive and supportive lab culture that I’m proud to be a part of.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="section-5" class="slide level2" data-background-image="img/acknowledgements-friends.svg" data-background-size="contain">
<h2></h2>
<aside class="notes">
<p>Next I’d like to thank my friends. You all helped keep me sane during grad school. We made sure to have lots of fun and maintain a healthy work-life balance in this crazy chapter of life.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="section-6" class="slide level2" data-background-image="img/acknowledgements-family.svg" data-background-size="contain">
<h2></h2>
<aside class="notes">
<p>Finally, I’d like to thank my family - my sisters Katie and Erika, my parents, and my boyfriend David. David has the uncanny ability to take any rainy day and turn it into a sunny one. My parents have been my cheerleaders from the very beginning, and they always encouraged me to pursue whatever I was interested in. Your support means everything to me.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section id="section-7" class="slide level2" data-visibility="uncounted">
<h2></h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/frontispiece.png" height="500"></p>
</figure>
</div>
<aside class="notes">
<p>Thank you all for your attention and I’ll take any questions now</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>








    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="presentation_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="presentation_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="presentation_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="presentation_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="presentation_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="presentation_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="presentation_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="presentation_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="presentation_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="presentation_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>